lib/ansible/modules/cloud/amazon/ec2_vpc_nat_gateway_facts.py

```diff
@@ -0,0 +1,147 @@
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+import json
+
+# import module snippets
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.ec2 import ec2_argument_spec, get_aws_connection_info, boto3_conn
+from ansible.module_utils.ec2 import ansible_dict_to_boto3_filter_list, HAS_BOTO3
+
+DOCUMENTATION = '''
+module: ec2_vpc_nat_gateway_facts
+short_description: Retrieves AWS VPC Managed Nat Gateway details using AWS methods.
+description:
+  - Gets various details related to AWS VPC Managed Nat Gateways
+version_added: "2.3"
+requirements: [ boto3 ]
+options:
+  nat_gateway_ids:
+    description:
+      - Get details of specific nat gateway IDs
+    required: false
+    default: None
+  filters:
+    description:
+      - A dict of filters to apply. Each dict item consists of a filter key and a filter value.
+        See U(http://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_DescribeNatGateways.html)
+        for possible filters.
+    required: false
+    default: None
+author: Karen Cheng(@Etherdaemon)
+extends_documentation_fragment:
+  - aws
+  - ec2
+'''
+
+EXAMPLES = '''
+# Simple example of listing all nat gateways
+- name: List all managed nat gateways in ap-southeast-2
+  ec2_vpc_nat_gateway_facts:
+    region: ap-southeast-2
+  register: all_ngws
+
+- name: Debugging the result
+  debug:
+    msg: "{{ all_ngws.result }}"
+
+- name: Get details on specific nat gateways
+  ec2_vpc_nat_gateway_facts:
+    nat_gateway_ids:
+      - nat-1234567891234567
+      - nat-7654321987654321
+    region: ap-southeast-2
+  register: specific_ngws
+
+- name: Get all nat gateways with specific filters
+  ec2_vpc_nat_gateway_facts:
+    region: ap-southeast-2
+    filters:
+      state: ['pending']
+  register: pending_ngws
+
+- name: Get nat gateways with specific filter
+  ec2_vpc_nat_gateway_facts:
+    region: ap-southeast-2
+    filters:
+      subnet-id: subnet-12345678
+      state: ['available']
+  register: existing_nat_gateways
+'''
+
+RETURN = '''
+result:
+  description: The result of the describe.
+    See http://boto3.readthedocs.io/en/latest/reference/services/ec2.html#EC2.Client.describe_nat_gateways for the response.
+  returned: success
+  type: list
+'''
+
+try:
+    import botocore
+except ImportError:
+    pass  # will be detected by imported HAS_BOTO3
+
+
+def date_handler(obj):
+    return obj.isoformat() if hasattr(obj, 'isoformat') else obj
+
+
+def get_nat_gateways(client, module, nat_gateway_id=None):
+    params = dict()
+
+    params['Filter'] = ansible_dict_to_boto3_filter_list(module.params.get('filters'))
+    params['NatGatewayIds'] = module.params.get('nat_gateway_ids')
+
+    try:
+        result = json.loads(json.dumps(client.describe_nat_gateways(**params), default=date_handler))
+    except Exception as e:
+        module.fail_json(msg=str(e.message))
+
+    return result['NatGateways']
```

Comment:
You need to fix up the result so that it is snake_cased format using the ec2.py helper function https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/cloud/amazon/GUIDELINES.md#camel_dict_to_snake_dict